metadata-db-app/src/cmr/metadata_db/services/concept_service.clj

```diff
@@ -426,6 +426,30 @@
   (fn [context concept]
     (boolean (:deleted concept))))
 
+(defn- delete-associations
+  "Delete the associations associated with the given collection revision and association type,
+  no association deletion event is generated."
+  [context assoc-type coll-concept-id coll-revision-id]
+  (let [search-params (cutil/remove-nil-keys
+                       {:concept-type assoc-type
+                        :associated-concept-id coll-concept-id
+                        :associated-revision-id coll-revision-id
+                        :exclude-metadata true
+                        :latest true})
+        associations (search/find-concepts context search-params)]
+    (doseq [association associations]
+      (save-concept-revision context {:concept-type assoc-type
```

Comment:
I didn't realize we tombstoned associations - are we cleaning these up with a job?